src/index.js

```diff
@@ -1,85 +1,81 @@
-// values need to know about the schema for two reasons—they need
-// access to specific instances of node and mark types, and they need
-// to know which of the node and mark types that they know about are
-// actually present in the schema.
-//
-// The `exampleSetup` plugin ties these together into a plugin that
-// will automatically enable this basic functionality in an editor.
+export { MessageMarkdownTransformer } from './schema/markdown/messageParser';
+export { ArticleMarkdownTransformer } from './schema/markdown/articleParser';
 
-// :: (Object) → [Plugin]
-// A convenience plugin that bundles together a simple menu with basic
-// key bindings, input rules, and styling for the example schema.
-// Probably only useful for quickly setting up a passable
-// editor—you'll need more control over your settings in most
-// real-world situations.
-//
-//   options::- The following options are recognized:
-//
-//     schema:: Schema
-//     The schema to generate key bindings and menu items for.
-//
-//     mapKeys:: ?Object
-//     Can be used to [adjust](#example-setup.buildKeymap) the key bindings created.
-//
-//     menuBar:: ?bool
-//     Set to false to disable the menu bar.
-//
-//     history:: ?bool
-//     Set to false to disable the history plugin.
-//
-//     floatingMenu:: ?bool
-//     Set to false to make the menu bar non-floating.
-//
-//     menuContent:: [[MenuItem]]
-//     Can be used to override the menu content.
-export function wootWriterSetup(options) {
+export { ArticleMarkdownSerializer } from './schema/markdown/articleSerializer';
+export { MessageMarkdownSerializer } from './schema/markdown/messageSerializer';
+
+export { fullSchema } from './schema/article';
+export { messageSchema } from './schema/message';
+
+export function wootArticleWriterSetup(props) {
   let plugins = [
-    ...(options.plugins || []),
-    buildInputRules(options.schema),
-    keymap(buildKeymap(options.schema, options.mapKeys)),
-    keymap(baseKeymap),
+    history(),
+    baseKeyMaps(props.schema),
+    blocksInputRule(props.schema),
+    textFormattingInputRules(props.schema),
+    linksInputRules(props.schema),
+    listInputRules(props.schema),
     dropCursor(),
     gapCursor(),
-    Placeholder(options.placeholder),
+    Placeholder(props.placeholder),
+    menuBar({
+      floating: true,
+      content: buildArticleEditorMenu(props.schema).fullMenu,
+    }),
+    new Plugin({
+      props: {
+        attributes: { class: 'ProseMirror-woot-style' },
+      },
+    }),
+    ...(props.plugins || []),
   ];
-  if (options.menuBar !== false)
-    plugins.push(
-      menuBar({
-        floating: options.floatingMenu !== false,
-        content: options.menuContent || buildMenuItems(options.schema).fullMenu,
-      })
-    );
-  if (options.history !== false) plugins.push(history());
 
-  return plugins.concat(
+  return plugins;
+}
+
+export function wootMessageWriterSetup(props) {
```

Comment:
Creating two different plugin sets for message and article editor. So we could separate the schema and input rules(Keyboard shortcuts etc)